extensions/wikia/Recirculation/services/ParselyDataService.class.php

```diff
@@ -0,0 +1,197 @@
+		return $data;
+	}
+
+	/**
+	 * Make an API request to parsely to gather posts
+	 * @param string $type
+	 * @return an array of posts
+	 */
+	private function apiRequest( $type, $meta, $count ) {
+		$options = [];
+		$options['limit'] = $count + self::PARSELY_API_EXTRA;
+
+		switch ( $type ) {
+			case 'vertical':
+				$options['sort'] = self::PARSELY_API_SORT;
+				$options['pub_days'] = 30;
+				$endpoint = 'analytics/tag/' . rawurlencode( $meta['tag'] ) . '/detail';
+				break;
+			case 'community':
+				$options['sort'] = self::PARSELY_API_SORT;
+				$options['pub_days'] = 30;
+				$endpoint = 'analytics/tag/' . rawurlencode( $meta['tag'] ) . '/detail';
+				break;
+			case 'shares':
+				$options['days'] = 5;
+				$endpoint = 'shares/posts';
+				break;
+			case 'recent_popular':
+				$options['sort'] = self::PARSELY_API_SORT;
+				$options['pub_days'] = self::PARSELY_API_PUB_DAYS;
+				$endpoint = 'analytics/posts';
+				break;
+			case 'popular':
+			default:
+				$options['sort'] = self::PARSELY_API_SORT;
+				$endpoint = 'analytics/posts';
+				break;
+		}
+
+		$url = $this->buildUrl( $endpoint, $options );
+		$data = ExternalHttp::get( $url );
+
+		$posts = json_decode( $data, true );
+
+		if ( isset( $posts['data'] ) && is_array( $posts['data'] ) ) {
+			return $this->dedupePosts( $posts['data'], $count );
+		} else {
+			return [];
+		}
+	}
+
+	/**
+	 * Build a complete url to the parsely API
+	 * @param string $endpoint
+	 * @param array $options
+	 * @return string
+	 */
+	private function buildUrl( $endpoint, $options ) {
+		global $wgParselyApiKey, $wgParselyApiSecret;
+
+		$defaultParams = [
+			'apikey' => $wgParselyApiKey,
+			'secret' => $wgParselyApiSecret,
+			'page' => self::PARSELY_API_PAGE
+		];
+
+		$params = array_merge( $defaultParams, $options );
+
+		$url = self::PARSELY_API_BASE . $endpoint . '?' . http_build_query( $params );
+
+		return $url;
+	}
+
+	/**
+	 * Remove duplicates from Parsely's API result
+	 * @param array $rawPosts
+	 * @return array
+	 */
+	private function dedupePosts( $rawPosts, $count ) {
+		$posts = [];
+		$postIds = [];
+
+		foreach ( $rawPosts as $post ) {
+			if ( count( $posts ) >= $count ) {
+				break;
+			}
+
+			$metadata = json_decode( $post['metadata'] );
+			if ( !empty( $metadata->postID ) && !in_array( $metadata->postID, $postIds ) ) {
+				$postIds[] = $metadata->postID;
+				$post['source'] = 'fandom';
+				$posts[] = $post;
+			}
+		}
+
+		return $posts;
+	}
+
+	private function buildMetaData( $type ) {
+		if ( $type === 'vertical' ) {
+			$metadata = $this->buildVerticalData();
+		} elseif ( $type === 'community' ) {
+			$metadata = $this->buildCommunityData();
+		} else {
+			$metadata = [
+				'key' => '',
+				'tag' => ''
+			];
+		}
+
+		return $metadata;
+	}
+
+	private function buildVerticalData() {
+		global $wgCityId;
+		$wikiFactoryHub = WikiFactoryHub::getInstance();
+		$wgWikiVertical = $wikiFactoryHub->getWikiVertical( $wgCityId )['short'];
+
+		$verticalMap = [
```

Comment:
Not currently. It's a map between the vertical of the current community and the tag name of the vertical on Fandom.